client/src/components/profile-box.js

```diff
@@ -0,0 +1,225 @@
+import React, { useEffect } from 'react';
+import { Form, Icon, Input, Modal, Row, Col } from 'antd';
+import { AuthStore } from '../stores';
+import { inject } from '../utils';
+import 'draft-js-static-toolbar-plugin/lib/plugin.css';
+
+const ProfileBox = ({
+  stores: [authStore],
+  form: {
+    getFieldDecorator,
+    validateFields,
+    setFieldsValue,
+    getFieldValue,
+    getFieldsValue,
+  },
+  visible,
+  toggleModal,
+  initialProfile,
+}) => {
+  let setBaseInfo = () => {
+    let user = authStore.user;
+    if (!user) return;
+    Object.keys(getFieldsValue()).forEach(key => {
+      const obj = {};
+      obj[key] = user[key] || null;
+      setFieldsValue(obj);
+    });
+    setFieldsValue({
+      userName: authStore.user.username,
+    });
+  };
+  useEffect(() => {
+    setBaseInfo();
+  }, [authStore.user]);
+
+  // User cannot change the username and password
+  var username = authStore.user.username;
+  // User can change the following field
+  var displayName = authStore.user.displayName;
+  var email = authStore.user.email;
+  var bio = authStore.user.bio;
+  var github = authStore.user.github;
+  var host = authStore.user.host;
+  var firstName = authStore.user.firstName;
+  var lastName = authStore.user.lastName;
+  var url = authStore.user.url;
+
+  let onUpdate = async e => {
+    e.preventDefault();
+    validateFields(async (err, values) => {
+      if (!err) {
+        try {
+          email = getFieldValue('email');
```

Comment:
There is no point to destruct each field, all you need are in the `values` parameter.